### PR TITLE
Idea for resolution to Issue #868

### PR DESCRIPTION
**Pull request by AdamMTGreenberg**
If we look at the stack trace we see:
```
Caused by: java.lang.IndexOutOfBoundsException: Wrong offset: 29566. Should be in range: [0, 29472]
	at org.jetbrains.kotlin.com.intellij.openapi.editor.impl.LineSet.findLineIndex(LineSet.java:178)
	at org.jetbrains.kotlin.com.intellij.openapi.editor.impl.DocumentImpl.getLineNumber(DocumentImpl.java:991)
	at com.pinterest.ktlint.core.ast.PackageKt.lineNumber(package.kt:235)
	at com.pinterest.ktlint.ruleset.experimental.AnnotationRule.visit(AnnotationRule.kt:149)
	at com.pinterest.ktlint.core.KtLint$format$2.invoke(KtLint.kt:376)
```
which corresponds to `val lineNumber = node.lineNumber()` - this shouldn't cause an error, but it does, most likely because the `node` was modified before.

I might have been trying to do too much in the combination of the two `AnnotationRule` with the multi-line stuff so I split them up. I played around with the idea of just searching for whitespace instead of line numbers, but if we are truly dealing with an issue with other rules then they could modify the lines as well.

### CODE REVIEW

**Comment by AdamMTGreenberg** [comment]
Fix proposed for Issue #868

**Comment by AdamMTGreenberg** [comment]
Tested by creating a file `test.kt`:
```
package a.b.c

class Test {
    fun bloop() {
        asdfadsf(asdfadsf, asdfasdf, asdfasdfasdfads,
        asdfasdf, asdfasdf, asdfasdf)
    }

    @Blah
    val test: Int
}
```
and then running:
```
./gradlew shadowJarExecutable &&  ./ktlint/build/run/ktlint --experimental -F /path/to/test.kt`

Result:
```
package a.b.c

class Test {
    fun bloop() {
        asdfadsf(
            asdfadsf,
            asdfasdf,
            asdfasdfasdfads,
            asdfasdf,
            asdfasdf,
            asdfasdf
        )
    }

    @Blah
    val test: Int
}
```

**Comment by AdamMTGreenberg** [comment]
I was able to achieve the same results without needing the line numbers - so removed it's usage. My initial assumption about the reuse of `node` was largely incorrect.

However, feel that this should be split into its own Rule since it differs enough certain people might want it in its own class or ignore rule

File: ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/ExperimentalRuleSetProvider.kt [hunk 1]
```diff
@@ -3,7 +3,7 @@ package com.pinterest.ktlint.ruleset.experimental
 import com.pinterest.ktlint.core.RuleSet
 import com.pinterest.ktlint.core.RuleSetProvider
 
-class ExperimentalRuleSetProvider : RuleSetProvider {
+public class ExperimentalRuleSetProvider : RuleSetProvider {
```
Comment (shashachu) [inline ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/ExperimentalRuleSetProvider.kt:6]: Was this change necessary?

File: ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/MultiLineAnnotationRule.kt [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+import org.jetbrains.kotlin.psi.KtAnnotationEntry
+import org.jetbrains.kotlin.psi.psiUtil.children
+import org.jetbrains.kotlin.psi.psiUtil.endOffset
+
+/**
+ * Ensures there are not multiple line breaks between annotations and declarations.
+ *
+ * https://kotlinlang.org/docs/reference/coding-conventions.html#annotation-formatting
+ */
+class MultiLineAnnotationRule : Rule("multi-line-annotation") {
```
Comment (shashachu) [inline ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/MultiLineAnnotationRule.kt:24]: Maybe a more descriptive name would be 'annotation-spacing' or something similar?

File: ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/ExperimentalRuleSetProvider.kt (hunk 1 above)
Comment (AdamMTGreenberg) [reply ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/ExperimentalRuleSetProvider.kt:6]: Linter was complaining in the IDE that it needed an explicit `public` declaration

File: ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/MultiLineAnnotationRule.kt (hunk 2 above)
Comment (AdamMTGreenberg) [reply ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/MultiLineAnnotationRule.kt]: Updated to match your suggestions
